etl/parse_email_save_attachment/parse_messages.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python3
+
+import os
+import pytz
+import boto3
+import email
+import tempfile
+from email import policy
+from operator import itemgetter
+from datetime import datetime
+
+
+def get_most_recent_file(bucket_name, prefix):
+    s3 = boto3.client("s3")
+
+    # List objects within the specified bucket and prefix
+    response = s3.list_objects_v2(Bucket=bucket_name, Prefix=prefix)
+
+    # Check if the bucket is empty or the prefix doesn't match any files
+    if "Contents" not in response:
+        return "No files found in the bucket with the specified prefix."
+
+    # Sort the files by last modified date
+    files = sorted(response["Contents"], key=itemgetter("LastModified"), reverse=True)
+
+    # Get the most recent file
+    most_recent_file = files[0]
+    return most_recent_file["Key"]
+
+
+def get_file_content(bucket_name, file_key):
+    s3 = boto3.client("s3")
+
+    # Get the object
+    obj = s3.get_object(Bucket=bucket_name, Key=file_key)
+
+    # Get the object's content
+    content = obj["Body"].read().decode("utf-8")
+    return content
+
+
+def parse_email_from_s3(email_content):
+    # Decode the string into bytes
+    email_content_bytes = email_content.encode("utf-8")
+    parsed_email = email.message_from_bytes(email_content_bytes, policy=policy.default)
+
+    # Create a temporary directory to store attachments
+    temp_dir = tempfile.mkdtemp()
+    if parsed_email.is_multipart():
+        # If email has multiple parts (like attachments or different content types)
+        for part in parsed_email.iter_parts():
+            if part.is_attachment():
+                # Save the attachment in the temporary directory
+                filename = part.get_filename()
+                if filename:
+                    filepath = os.path.join(temp_dir, filename)
+                    with open(filepath, "wb") as f:
+                        f.write(part.get_payload(decode=True))
+
+    return temp_dir
+
+
+import glob
```

Comment:
Saw this import hanging out here - could it be moved either up to the top of the file or inside `upload_attachments_to_s3`?

Reply:
100% yes - thank you! This got moved up and organized in the imports where it belonged. :pray: